### PR TITLE
Assign contacts to campaign

### PR DESCRIPTION
**Pull request by JuanCarlosMartinez4**

### CODE REVIEW

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 1]
```diff
@@ -158,6 +164,17 @@ public static NotesAndAttachmentsPageAbstract getNotesAndAttachments() {
         return new NotesAndAttachmentsLightningPage();
     }
 
+    /**
+     * Allows to indentify campaign members user experience page.
+     * @return campaign members instance.
+     */
+    public static AbstractCampaignMembersPage getCampaignMembers() {
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java]: ctrol + alt  +L

File: src/main/java/salesforce/ui/pages/campaign/AbstractCampaignPage.java [hunk 2]
```diff
@@ -31,4 +33,15 @@ protected void waitUntilPageObjectIsLoaded() {
      */
     public abstract AbstractCampaignListPage deleteCampaign();
 
+    /**
+     * Adds campaign members.
+     * @return campaign contact instance.
+     */
+    public abstract AbstractCampaignContactPage addCampaignMembers();
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/campaign/AbstractCampaignPage.java]: ctrl + alt + L

File: src/main/java/salesforce/ui/pages/campaigncontact/AbstractCampaignContactPage.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Manages campaign contacts.
+ *
+ * @author Juan Martinez.
+ * @version 1.0 25 March 2020.
+ */
+public abstract class AbstractCampaignContactPage extends AbstractBasePage {
+
+    /**
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/campaigncontact/AbstractCampaignContactPage.java:24]: ctrl + alt + L

File: src/main/java/salesforce/ui/pages/campaigncontact/CampaignContactClassicPage.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+     * @return selector composed.
+     */
+    private Select getSelect(final String xpathLocator) {
+        return new Select(webDriver.findElement(By.xpath(xpathLocator)));
+    }
+
+    /**
+     * Selects item in combobox.
+     * @param xpathCombobox value.
+     * @param viewName value.
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/campaigncontact/CampaignContactClassicPage.java]: ctrl + alt + L

File: src/main/java/salesforce/ui/pages/campaigncontact/CampaignContactClassicPage.java [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        for (String contactName : contactsNames) {
+            checkContact(contactName);
+        }
+    }
+
+    /**
+     * Generates a list of contacts.
+     * @param namesContacts list.
+     * @return contacts names list.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/campaigncontact/CampaignContactClassicPage.java]: ctrl + alt + L

File: src/main/java/salesforce/ui/pages/campaigncontact/CampaignContactPopupLightningPage.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    private void clickOnAddContactsPopupTitle() {
+        webDriverWait.until(ExpectedConditions.visibilityOf(addContactsPopupTitle));
+        addContactsPopupTitle.click();
+    }
+
+    /**
+     * Gets web element.
+     * @param contactName value.
+     * @return web element.
+     */
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/campaigncontact/CampaignContactPopupLightningPage.java]: ctrl + alt + L

File: src/main/java/salesforce/ui/pages/newcampaign/NewCampaignLightningPopUp.java [hunk 7]
```diff
@@ -31,8 +30,6 @@
 
     @Override
     protected void waitUntilPageObjectIsLoaded() {
-        webDriverWait.until(ExpectedConditions.visibilityOf(campaignNameField));
-        webDriverWait.until(ExpectedConditions.elementToBeClickable(campaignNameField));
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/newcampaign/NewCampaignLightningPopUp.java:35]: Why did you delete these?

File: src/main/java/salesforce/utils/WebElementChecker.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+package salesforce.utils;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+/**
+ * Checks if web element is displayed.
+ *
+ * @author Juan Martinez.
+ * * @version 1.0 25 March 2020.
```
Comment (Amet015) [inline src/main/java/salesforce/utils/WebElementChecker.java]: review this  **

File: src/test/resources/features/PriceBooks.feature [hunk 9]
```diff
@@ -1,11 +1,11 @@
...
\ No newline at end of file
+#  @wip
+#  Scenario: Assign products to a new Price Book
+#    Given I login as "Admin"
+#    And I create product as ProductTest01
+#    And I create product as ProductTest02
+#    And I create PriceBook as "PriceBookTest"
+#    And I search "PriceBookTest"
+#    When To do
+#    Then To do
```
Comment (Amet015) [inline src/test/resources/features/PriceBooks.feature]: add new line here

File: src/main/java/salesforce/ui/pages/newcampaign/NewCampaignLightningPopUp.java (hunk 7 above)
Comment (JuanCarlosMartinez4) [reply src/main/java/salesforce/ui/pages/newcampaign/NewCampaignLightningPopUp.java:35]: because that locator appears, but sometimes that locator is not reconigzed

File: src/main/java/salesforce/ui/pages/AppPageFactory.java (hunk 1 above)
Comment (Amet015) [reply src/main/java/salesforce/ui/pages/AppPageFactory.java]: x2

File: src/main/java/salesforce/ui/pages/AppPageFactory.java [hunk 10]
```diff
@@ -158,6 +164,17 @@ public static NotesAndAttachmentsPageAbstract getNotesAndAttachments() {
...
+        if (userExperience.equals("Classic")) {
+            return new CampaignMembersClassicPage();
+        }
+        return new CampaignMembersLightningPage();
+    }
+
     /**
```
Comment (Amet015) [inline src/main/java/salesforce/ui/pages/AppPageFactory.java:179]: ctrol + alt +L

File: src/test/java/salesforcetest/steps/CampaignSteps.java [hunk 11]
```diff
@@ -9,31 +9,87 @@
...
+    private AbstractCampaignContactPage campaignContact;
     private PageTransporter pageTransporter;
     private static final int ARRAY_POSITION_FIRST = 0;
 
     /**
-     * OpportunityStep constructor.
-     *
-     * @param context value.
+     * Constructor of campaigns steps.
+     * @param context instance.
```
Comment (Amet015) [inline src/test/java/salesforcetest/steps/CampaignSteps.java]: ctrl + alt + L

File: src/test/java/salesforcetest/steps/CampaignSteps.java [hunk 12]
```diff
@@ -9,31 +9,87 @@
...
     public CampaignSteps(final Context context) {
         this.context = context;
+        contacts = context.getContacts();
         pageTransporter = new PageTransporter();
     }
 
+    /**
+     * Selects a campaign by name.
+     * @param campaignName value.
+     */
```
Comment (Amet015) [inline src/test/java/salesforcetest/steps/CampaignSteps.java]: ctrl + alt + L

**Review by Amet015 — APPROVED** [review]
